### PR TITLE
Improve Airflow logging for operator Jinja template processing

### LINKED ISSUE #25344: Improve Airflow logging for operator Jinja template processing
### Description

When an operator uses Jinja templating, debugging issues is difficult because the Airflow task log only displays a stack trace.

### Use case/motivation

When there's a templating issue, I'd like to have some specific, actionable info to help understand the problem. At minimum:
* Which operator or task had the issue?
* Which field had the issue?
* What was the Jinja template?

Possibly also the Jinja context, although that can be very verbose.

I have prototyped this in my local Airflow dev environment, and I propose something like the following. (Note the logging commands, which are not present in the Airflow repo.)

Please let me know if this sounds reasonable, and I will be happy to create a PR.

```
def _do_render_template_fields(
    self,
    parent,
    template_fields,
    context,
    jinja_env,
    seen_oids,
) -> None:
    """Copied from Airflow 2.2.5 with added logging."""
    logger.info(f"BaseOperator._do_render_template_fields(): Task {self.task_id}")
    for attr_name in template_fields:
        content = getattr(parent, attr_name)
        if content:
            logger.info(f"Rendering template for '{attr_name}' field: {content!r}")
            rendered_content = self.render_template(content, context, jinja_env, seen_oids)
+            setattr(parent, attr_name, rendered_content)
```

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by barrywhart**
closes: #25344

Adds logging to provide debugging help for failures in `render_template_fields()`

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/models/abstractoperator.py [hunk 1]
```diff
@@ -77,6 +78,9 @@
 )
 
 
+logger = logging.getLogger("airflow.models.abstractoperator.AbstractOperator")
```
Comment (barrywhart) [inline airflow/models/abstractoperator.py]: Added logger instance, following the same pattern as `baseoperator.py`.

File: airflow/models/abstractoperator.py [hunk 2]
```diff
@@ -391,8 +404,8 @@ def render_template(
                 template = jinja_env.from_string(value)
             dag = self.get_dag()
             if dag and dag.render_template_as_native_obj:
-                return render_template_as_native(template, context)
-            return render_template_to_string(template, context)
+                return render_template_as_native(template, context)  # type: ignore[arg-type]
+            return render_template_to_string(template, context)  # type: ignore[arg-type]
```
Comment (barrywhart) [inline airflow/models/abstractoperator.py]: Work around `mypy` failures. These issues seemed to be unrelated to my changes.

File: airflow/models/abstractoperator.py [hunk 3]
```diff
@@ -345,13 +349,22 @@ def _do_render_template_fields(
...
+                    context,
+                    jinja_env,
+                    seen_oids,
+                )
+            except Exception:
+                logger.exception(
+                    f"Exception rendering Jinja template for in "
+                    f"task '{self.task_id}', field "
+                    f"'{attr_name}'. Template: {value!r}"
+                )
```
Comment (uranusjr) [inline airflow/models/abstractoperator.py]: I think `self.log.exception` is better here; `baseoperator` needed an additional `logger` for free functions outside a class, but this inherits `LoggingMixin` and does not need it.

File: airflow/models/abstractoperator.py (hunk 2 above)
Comment (uranusjr) [reply airflow/models/abstractoperator.py]: What are the errors? We do not want to introduce necessary ignores into the code base; if the errors are indeed unrelated, they need to be fixed first.

Comment (barrywhart) [reply airflow/models/abstractoperator.py]: These are the errors. When you say:
>they need to be fixed first.

Are you requesting that I fix them, or someone else will fix them?

```
airflow/models/abstractoperator.py:402: error: Argument 2 to "render_template_as_native" has incompatible type "Context"; expected "MutableMapping[str, Any]"  [arg-type]
                    return render_template_as_native(template, context)
                                                               ^
airflow/models/abstractoperator.py:403: error: Argument 2 to "render_template_to_string" has incompatible type "Context"; expected "MutableMapping[str, Any]"  [arg-type]
                return render_template_to_string(template, context)
```

File: airflow/models/abstractoperator.py (hunk 3 above)
Comment (barrywhart) [reply airflow/models/abstractoperator.py]: Done!

File: airflow/models/abstractoperator.py (hunk 2 above)
Comment (barrywhart) [reply airflow/models/abstractoperator.py]: I removed the `# type: ignore` comments. We'll see if the CI requires them. Local `mypy` didn't like those lines.

For what it's worth, these are the versions of mypy and related packages in my Breeze Docker environment (newly set up yesterday):
```
mypy==0.950
mypy-boto3-appflow==1.24.36.post1
mypy-boto3-rds==1.24.38
mypy-boto3-redshift-data==1.24.36.post1
mypy-extensions==0.4.3
```

**Comment by barrywhart** [comment]
Updated and ready for another review!

**Review by potiuk — APPROVED** [review]
COOL!

**Pull request merged**